easybuild/tools/modules.py

```diff
@@ -1433,30 +1457,12 @@ def use(self, path, priority=None):
             if os.environ.get('__LMOD_Priority_MODULEPATH'):
                 self.run_module(['use', path])
             else:
-                cur_mod_path = os.environ.get('MODULEPATH')
-                if cur_mod_path is None:
-                    new_mod_path = path
-                else:
-                    new_mod_path = [path] + [p for p in cur_mod_path.split(':') if p != path]
-                    new_mod_path = ':'.join(new_mod_path)
-                self.log.debug('Changing MODULEPATH from %s to %s' %
-                               ('<unset>' if cur_mod_path is None else cur_mod_path, new_mod_path))
-                os.environ['MODULEPATH'] = new_mod_path
+                self._set_module_path([path] + [p for p in curr_module_paths(clean=False) if p != path])
 
     def unuse(self, path):
         """Remove a module path"""
         # We can simply remove the path from MODULEPATH to avoid the costly module call
```

Comment:
`costly "module unuse" call`

Maybe also clarify that modifying `$MODULEPATH` directly implies that Lmod won't reconsider loaded modules, but in the context of an EasyBuild session that should be OK...

Maybe we should also add an option to force running the actual `module unuse` command (same above for `use`), in case people are using the EasyBuild interface to the modules tool in Python scripts?

Reply:
Done with default set to False to provide the fast implementation by default